README.md

```diff
@@ -1,6 +1,218 @@
+ an overview of the system.
+
+### What It Does
+This broker implements OpenServiceBroker methods for creating, connecting to, and destroying
+object buckets.
+- For each new Service Instance, a new, unquiely named bucket is created.
+- For each new Service Instance Credential, a Secret is generated with the
+coordinates and credentials of the Service Instance's associated bucket.
+- Deleting a Service Instance destroys the associated bucket.
+- Deleting a Service Instance Credential deletes the secret.  Nothing is done on the broker side.  This is because the broker does not perform any actions on Bind, so there is nothing to
+undo.
+
+## Installation
+
+### Dependencies
+- [Google SDK](https://cloud.google.com/sdk/) (`gcloud`): To access GCP via cli.
+- [Kubernetes](https://github.com/kubernetes/kubernetes): to run local K8s cluster
+- [Gluster-Kubernetes Deploy Script](https://github.com/copejon/gk-cluster-deploy): to deploy GCP K8s cluster with Gluster-Kubernetes Object Storage and CNS Object Broker
+- Kubernetes-Incubator [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog): to provide Service-Catalog helm chart
+- Kubernetes [Helm](https://github.com/kubernetes/helm#install): to install Service-Catalog chart
+
+### Assumptions
+- Gk-cluster-deploy was written and tested on Fedora 25 and RHEL 7.
+- A [Google Cloud Platform](https://cloud.google.com/compute/) account.
+
+### Topology
+There are two primary systems that make up this demonstration.  They are the Broker
+and its colocated CNS Object store.  They fill the role of our micro-service provider.
+A second system will be the locally running Kubernetes cluster on which with Service-Catalog is deployed.  This cluster will be our micro-server client.  Please refer to the [command flow diagram](./docs/Service-Catalog-Published.png) for a more in depth look at these systems.
+
+## Setup
+### Step 0: Preparing environment
+- Clone [Kubernetes](https://github.com/kubernetes)
+- Clone [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog)
+- Install [Google SDK](https://cloud.google.com/sdk/) and add `gcloud` to `PATH`.  You must have an active Google Cloud Platform account.
+- Configure `gcloud` user, zone, and project.  These are detected by the deployment script and required for setup.
+
+
+`# gcloud config set account <user account>`
+
+`# gcloud config set project <project>`
+
+`# gcloud config set compute/zone <zone>`
+
+
+### Step 1: Deploy Gluster-Kubernetes cluster in Google Compute Engine (GCE)
+This step sets up the **External Service Provider** portion of our topology (see [diagram](./docs/Service-Catalog-Published.png)).
+
+To kick off deployment, run `gk-cluster-deploy/deploy/cluster/gk-up.sh`.  The script has a number of configurable
+variables relative to GCE Instance settings.  These can be defined inline with `gk-up.sh` or as environment variables.  To run the script as fire and forget, pass `-y` to skip configuration review.
```

Comment:
"... or as environment variables ..."  <-- Add: "The [config.sh](https://github.com/copejon/gk-cluster-deploy/blob/master/deploy/cluster/lib/config.sh)" config script contains all of the config-related variables used by `gk-up.sh`.